### PR TITLE
[GEOS-9146] Add REST API to configure the params-extractor community module

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/EchoesController.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        MediaTypeExtensions.TEXT_JSON_VALUE
+    }
+)
+public class EchoesController extends RestBaseController {
+    static final String ECHOES_ROOT = RestBaseController.ROOT_PATH + "/params-extractor/echoes";
+
+    @GetMapping
+    public RestListWrapper<EchoParameter> getEchoParameters() {
+        return new RestListWrapper<>(
+                EchoParametersDao.getEchoParameters().stream().collect(Collectors.toList()),
```
Comment (nmco) [inline src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/EchoesController.java]: `EchoParametersDao.getEchoParameters()` already returns a list.

File: src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/RulesController.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        MediaTypeExtensions.TEXT_JSON_VALUE
+    }
+)
+public class RulesController extends RestBaseController {
+    static final String RULES_ROOT = RestBaseController.ROOT_PATH + "/params-extractor/rules";
+
+    @GetMapping
+    public RestListWrapper<Rule> getRules() {
+        return new RestListWrapper<>(
+                RulesDao.getRules().stream().collect(Collectors.toList()),
```
Comment (nmco) [inline src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/RulesController.java]: `RulesDao.getRules()` already returns a list.

File: src/community/params-extractor/src/test/java/org/geoserver/params/extractor/rest/RulesControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ */
```
Comment (nmco) [inline src/community/params-extractor/src/test/java/org/geoserver/params/extractor/rest/RulesControllerTest.java]: Wrong copyright header.

File: src/community/params-extractor/src/main/java/org/geoserver/params/extractor/RulesDao.java [hunk 4]
```diff
@@ -6,34 +6,32 @@
...
-import javax.xml.stream.XMLStreamWriter;
 import org.geoserver.config.GeoServerDataDirectory;
+import org.geoserver.config.util.SecureXStream;
 import org.geoserver.platform.GeoServerExtensions;
 import org.geoserver.platform.resource.Resource;
 import org.geotools.util.logging.Logging;
-import org.xml.sax.Attributes;
-import org.xml.sax.SAXException;
-import org.xml.sax.helpers.DefaultHandler;
+import org.h2.bnf.RuleList;
```
Comment (nmco) [inline src/community/params-extractor/src/main/java/org/geoserver/params/extractor/RulesDao.java]: Unused import.

File: src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/EchoesController.java (hunk 1 above)
Comment (aaime) [reply src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/EchoesController.java]: It's not going to be encoded by XStream with all the backlinks generation without wrapping in RestListWrapper.

File: src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/RulesController.java (hunk 2 above)
Comment (aaime) [reply src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/RulesController.java]: Same as above.

File: src/community/params-extractor/src/test/java/org/geoserver/params/extractor/rest/RulesControllerTest.java (hunk 3 above)
Comment (aaime) [reply src/community/params-extractor/src/test/java/org/geoserver/params/extractor/rest/RulesControllerTest.java]: Thanks, will fix!

File: src/community/params-extractor/src/main/java/org/geoserver/params/extractor/RulesDao.java (hunk 4 above)
Comment (aaime) [reply src/community/params-extractor/src/main/java/org/geoserver/params/extractor/RulesDao.java]: Going to remove

File: src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/EchoesController.java (hunk 1 above)
Comment (aaime) [reply src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/EchoesController.java]: Ah, the stream thing.. yes, I was doing map to another type before, and forgot to remove the entire thing.

Comment (nmco) [reply src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/EchoesController.java]: Sorry I was not clear, I was referring to:
```
EchoParametersDao.getEchoParameters().stream().collect(Collectors.toList())
```
... we are converting a list to a list.

**Pull request merged**